src/index.js

```diff
@@ -7,7 +7,7 @@ export function klona(x) {
 		if (x.constructor !== Object && typeof x.constructor === 'function') {
 			tmp = new x.constructor();
 			for (k in x) {
-				if (tmp.hasOwnProperty(k) && tmp[k] !== x[k]) {
+				if (x.hasOwnProperty(k) && tmp[k] !== x[k]) {
```

Comment:
This is redundant. You can't `for (key in thing)` and then have `thing.hasOwnProperty(key)` be false.

Reply:
I had removed the hasOwnProperty, but it ended up copying "hidden" properties.
So i find this answer:
https://stackoverflow.com/a/728694

Reply:
`for (key in thing)` returns all keys including the `prototype` thats why we use hasOwnProperty.

> Unlike the `in` operator, this method does not check for the specified property in the object's prototype chain.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/HasOwnProperty

Reply:
Additional information
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Enumerability_and_ownership_of_properties#detection_table

Reply:
If you want it is possible to use `for (key of Reflect.ownKeys(x))` instead of `for (key in thing)`

Reply:
@lukeed, can you  review it again?

Reply:
@lukeed i think its possible to use `for (key of Object.keys(thing)` too